examples/libh2o/httpclient.c

```diff
@@ -229,7 +229,9 @@ h2o_httpclient_head_cb on_connect(h2o_httpclient_t *client, const char *errstr,
                                   h2o_url_t *origin)
 {
     if (errstr != NULL) {
-        on_error(client->ctx, errstr);
+        char buf[128];
+        on_error(client->ctx, "%s:%s", errstr,
+                 h2o_strerror_r(errno, buf, sizeof(buf)));
```

Comment:
I do not think that we should be using `h2o_strerror_r` in the HTTP-level callbacks. There is no guarantee that the cause of `on_connect` failing is due to an error returned from a syscall (consider TLS certificate mismatch for an example).

Reply:
Yes, `on_connect` callback will be delayed after ssl handshake finished(or error) when using ssl for http layer.
How about do this in socketpool's on `on_connect`?

Reply:
That would collide with the current design, that uses the address of `errstr` to see what the error was.

The socket layer is expected define it's own error strings, and declare them as `const char *h2o_socket_error_XXX`. Applications are expected to compare the address of the returned string if it needs to take action based on the specific error that happened.

Reply:
Thanks for your clarification.
Seems currently applications just do the `NULL` judge and do not compare with socket layer error strings.
Anyway, we can leave it as it is.